### ISSUE TITLE
`Callable` should not be a valid type for `type[T]`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6868]
The following should generate an error but does not.

```python
def func[T](x: type[T]):
    ...

func(Callable)  # Should be an error
```

**Comment by ikamensh** [comment]
Out of curiosity, what is Callable if it's not valid type[T]? you ofcourse can't instantiate Callable like you can "proper" types. So Callable is a Protocol?

**Comment by erictraut** [comment]
`type` refers to the class that is the base class for the metaclass hierarchy. `Callable` is a special form. 

See [this part of the typing spec](https://typing.readthedocs.io/en/latest/spec/special-types.html#type) for details. In particular, it says:

> Any other special constructs like ... Callable are not allowed as an argument to type.

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
